frontends/web/src/components/sidebar/sidebar.tsx

```diff
@@ -90,207 +89,194 @@ const GetAccountLink = ({ coinCode, code, name, handleSidebarItemClick }: TGetAc
+      document.removeEventListener('touchend', handleTouchEnd);
     };
-  };
+  }, [sidebarStatus]);
 
-  private handleTouchMove = (e: TouchEvent) => {
-    if (this.props.sidebarStatus !== 'forceHidden') {
-      if (e.changedTouches && e.changedTouches.length) {
-        this.swipe.active = true;
-      }
-    }
-  };
+  const [keystores, setKeystores] = useState<TKeystores>();
 
-  private handleTouchEnd = (e: TouchEvent) => {
-    if (this.props.sidebarStatus !== 'forceHidden') {
-      const touch = e.changedTouches[0];
-      const travelX = Math.abs(touch.clientX - this.swipe.x);
-      const travelY = Math.abs(touch.clientY - this.swipe.y);
-      const validSwipe = window.innerWidth <= 901 && this.swipe.active && travelY < 100 && travelX > 70;
-      if ((!panelStore.state.activeSidebar && validSwipe && this.swipe.x < 60) ||
-                (panelStore.state.activeSidebar && validSwipe && this.swipe.x > 230)) {
-        toggleSidebar();
-      }
-      this.swipe = {
-        x: 0,
-        y: 0,
-        active: false,
-      };
-    }
-  };
+  useEffect(() => {
+    getKeystores().then(keystores => {
+      setKeystores(keystores);
+
+    });
+    return subscribeKeystores(setKeystores);
+  }, []);
 
-  private handleSidebarItemClick = (e: React.SyntheticEvent) => {
-    const el = (e.target as Element).closest('a');
+  const handleSidebarItemClick = (event: React.SyntheticEvent) => {
+    const el = (event.target as Element).closest('a');
     if (el!.classList.contains('sidebarActive') && window.innerWidth <= 901) {
       toggleSidebar();
     }
   };
 
-  public render() {
-    const { keystores } = this.state;
-    const {
-      t,
-      deviceIDs,
-      accounts,
-      activeSidebar,
-      sidebarStatus,
-    } = this.props;
-    const hidden = sidebarStatus === 'forceHidden';
-    const hasOnlyBTCAccounts = accounts.every(({ coinCode }) => isBitcoinOnly(coinCode));
-    const accountsByKeystore = getAccountsByKeystore(accounts);
-    return (
-      <div className={[style.sidebarContainer, hidden ? style.forceHide : ''].join(' ')}>
-        <div className={[style.sidebarOverlay, activeSidebar ? style.active : ''].join(' ')} onClick={toggleSidebar}></div>
-        <nav className={[style.sidebar, activeSidebar ? style.forceShow : ''].join(' ')}>
-          <div className={style.sidebarLogoContainer}>
-            <Link
-              to={accounts.length ? '/account-summary' : '/'}
-              onClick={this.handleSidebarItemClick}>
-              <AppLogoInverted className={style.sidebarLogo} />
-            </Link>
-            <button className={style.closeButton} onClick={toggleSidebar}>
-              <CloseXWhite />
-            </button>
-          </div>
+  const hidden = sidebarStatus === 'forceHidden';
+  const hasOnlyBTCAccounts = accounts.every(({ coinCode }) => isBitcoinOnly(coinCode));
+  const accountsByKeystore = getAccountsByKeystore(accounts);
```

Comment:
Yes these lines are currently executed on every render, like before this commit (those lines were just in render function).

But there is some potential for optimization here, as `hasOnlyBTCAccounts` and `accountsByKeystore` only depend on `accounts`. So we probably should move this so that `accounts.every(` and `getAccountsByKeystore(accounts)` only is executed when `accounts` change. This way it would not do the operations on every render.